service/history/mutableStateBuilder.go

```diff
@@ -572,48 +572,21 @@ func (e *mutableStateBuilder) GetCompletionEvent() (*workflow.HistoryEvent, bool
 }
 
 // DeletePendingChildExecution deletes details about a ChildExecutionInfo.
-func (e *mutableStateBuilder) DeletePendingChildExecution(initiatedEventID int64) error {
-	_, ok := e.pendingChildExecutionInfoIDs[initiatedEventID]
```

Comment:
why is it safe to remove those checks?

Reply:
This was dead code even before my change.  All the places it is called from, already checks existence of ChildInfo before calling this API.